scene/resources/animation.cpp

```diff
@@ -5594,7 +5594,10 @@ Variant Animation::array_to_string(const Variant p_value) {
 }
 
 Variant Animation::add_variant(const Variant &a, const Variant &b) {
-	if (a.get_type() != b.get_type() && !a.is_array()) {
+	if (a.is_num() && b.is_num()) {
+		const real_t result = (a.operator real_t()) + (b.operator real_t());
```

Comment:
I don't think this should be `real_t`, `Variant` always holds a `double`, there's some confusion over this at times and some times `real_t` is used incorrectly, this was fixed in some places here in #81296

Reply:
Oh, right, I just copied this from the `FLOAT` case without questioning it but I completely agree. `real_t` is likely to be `float` whereas the value in the `Variant` is always a `double`. I've changed the "float only" cases, too. Cubic interpolation also casts its `FLOAT` `Variants` to `real_t`. Not sure if we should change that, too.

Reply:
I'd say to take it one step at a time unless an animation reviewer says otherwise 🙂

Reply:
Actually, from looking at the PR you linked, I learned that `VARIANT::INT` is supposed to be `int64_t`. In the current code, it is interpolated via `real_t` and then cast to an `int`, so I expect the problem that was fixed there is back. I suppose that warrants a new pull request for fixing issue #81289 again?

Reply:
I'd say that should be handled here specifically then